src/catalog/trigger_catalog.cpp

```diff
@@ -0,0 +1,302 @@
+        LOG_INFO("trigger_name is %s", (*result_tiles)[i]->GetValue(j, 0).ToString().c_str());
+        LOG_INFO("FIRE_CONDITION is %s", (*result_tiles)[i]->GetValue(j, 1).ToString().c_str());
+        LOG_INFO("FUNCTION_NAME is %s", (*result_tiles)[i]->GetValue(j, 2).ToString().c_str());
+        LOG_INFO("FUNCTION_ARGS is %s", (*result_tiles)[i]->GetValue(j, 3).ToString().c_str());
+        LOG_INFO("reach here safely");
+        // create a new trigger instance
+        commands::Trigger newTrigger((*result_tiles)[i]->GetValue(j, 0).ToString(),
+                                     trigger_type,
+                                     (*result_tiles)[i]->GetValue(j, 2).ToString(),
+                                     (*result_tiles)[i]->GetValue(j, 3).ToString(),
+                                     (*result_tiles)[i]->GetValue(j, 1).ToString());
+        commands::TriggerList tmpTriggerList;
+        tmpTriggerList.AddTrigger(newTrigger);
+
+        LOG_INFO("reach here safely 2");
+        if (newTriggerList == nullptr) {
+          LOG_INFO("newTriggerList is null....");
+        }
+        newTriggerList->AddTrigger(newTrigger);
+        LOG_INFO("reach here safely 3");
+      }
+    }
+  }
+  LOG_INFO("reach here safely 4");
+  return newTriggerList;
+}
+
+commands::TriggerList* TriggerCatalog::GetTriggers(oid_t database_oid,
+                                          oid_t table_oid,
+                                          concurrency::Transaction *txn) {
+  LOG_INFO("get triggers for table %d", table_oid);
+  // select trigger_name, fire condition, function_name, function_args
+  std::vector<oid_t> column_ids({ColumnId::TRIGGER_NAME, ColumnId::TRIGGER_TYPE, ColumnId::FIRE_CONDITION, ColumnId::FUNCTION_NAME, ColumnId::FUNCTION_ARGS});
+  oid_t index_offset = IndexId::DATABASE_TABLE_KEY_1; // multi-column (database_oid, table_oid) bwtree index
+  std::vector<type::Value> values;
+  // where database_oid = args.database_oid and table_oid = args.table_oid and trigger_type = args.trigger_type
+  values.push_back(type::ValueFactory::GetIntegerValue(database_oid).Copy());
+  values.push_back(type::ValueFactory::GetIntegerValue(table_oid).Copy());
+
+  // the result is a vector of executor::LogicalTile
+  auto result_tiles =
+      GetResultWithIndexScan(column_ids, index_offset, values, txn);
+  // carefull! the result tile could be null!
+  if (result_tiles == nullptr) {
+    LOG_INFO("no trigger on table %d", table_oid);
+  } else {
+    LOG_INFO("size of the result tiles = %lu", result_tiles->size());
+  }
+
+  // create the trigger list
+  commands::TriggerList* newTriggerList = new commands::TriggerList();
+  if (result_tiles != nullptr) {
+    for (unsigned int i = 0; i < result_tiles->size(); i++) {
+      size_t tuple_count = (*result_tiles)[i]->GetTupleCount();
+      for (size_t j = 0; j < tuple_count; j++) {
+        LOG_INFO("trigger_name is %s", (*result_tiles)[i]->GetValue(j, 0).ToString().c_str());
```

Comment:
Is it necessary to have these LOG_INFO in final version of code? Will it have too much print out? Also, is LOG_DEBUG better?